From ec29ea82fd8e06ddebffd9c1dbdb53ec6872c3f2 Mon Sep 17 00:00:00 2001
From: David Rosca <nowrep@gmail.com>
Date: Sat, 13 Dec 2014 12:57:15 +0100
Subject: [PATCH] Adapter::name() now returns alias of adapter

According to Bluez docs, Alias property should be used
as a name for adapter. If no alias for adapter was set,
it returns original system name.

Added new method Adapter::systemName() that returns system
name (hostname) of the adapter.
---
 bluedevil/bluedeviladapter.cpp | 12 ++++++------
 bluedevil/bluedeviladapter.h   | 17 +++++++++++------
 2 files changed, 17 insertions(+), 12 deletions(-)

diff --git a/bluedevil/bluedeviladapter.cpp b/bluedevil/bluedeviladapter.cpp
index 668934a..e8d0abf 100644
--- a/bluedevil/bluedeviladapter.cpp
+++ b/bluedevil/bluedeviladapter.cpp
@@ -89,7 +89,7 @@ void Adapter::Private::_k_propertyChanged(const QString &interface_name, const Q
     for(i = changed_properties.constBegin(); i != changed_properties.constEnd(); ++i) {
       QVariant value = i.value();
       QString property = i.key();
-      if (property == "Name") {
+      if (property == "Alias") {
           emit m_q->nameChanged(value.toString());
       } else if (property == "Powered") {
           emit m_q->poweredChanged(value.toBool());
@@ -141,12 +141,12 @@ QString Adapter::address() const
 
 QString Adapter::name() const
 {
-    return d->m_bluezAdapterInterface->name();
+    return d->m_bluezAdapterInterface->alias();
 }
 
-QString Adapter::alias() const
+QString Adapter::systemName() const
 {
-    return d->m_bluezAdapterInterface->alias();
+    return d->m_bluezAdapterInterface->name();
 }
 
 quint32 Adapter::adapterClass() const
@@ -214,9 +214,9 @@ QStringList Adapter::UUIDs()
     return UUIDs;
 }
 
-void Adapter::setAlias(const QString& alias)
+void Adapter::setName(const QString& name)
 {
-    d->m_bluezAdapterInterface->setAlias(alias);
+    d->m_bluezAdapterInterface->setAlias(name);
 }
 
 void Adapter::setPowered(bool powered)
diff --git a/bluedevil/bluedeviladapter.h b/bluedevil/bluedeviladapter.h
index ccb17f9..11cd211 100644
--- a/bluedevil/bluedeviladapter.h
+++ b/bluedevil/bluedeviladapter.h
@@ -49,8 +49,8 @@ class BLUEDEVIL_EXPORT Adapter
     Q_OBJECT
 
     Q_PROPERTY(QString address READ address)
-    Q_PROPERTY(QString name READ name)
-    Q_PROPERTY(QString alias READ alias WRITE setAlias)
+    Q_PROPERTY(QString name READ name WRITE setName)
+    Q_PROPERTY(QString systemName READ systemName)
     Q_PROPERTY(quint32 adapterClass READ adapterClass)
     Q_PROPERTY(bool powered READ isPowered WRITE setPowered)
     Q_PROPERTY(bool discoverable READ isDiscoverable WRITE setDiscoverable)
@@ -75,14 +75,19 @@ public:
     QString address() const;
 
     /**
+     * Returns the friendly name of the adapter.
+     *
+     * Friendly name is alias of the adapter. If no alias is set, it returns
+     * the system name.
+     *
      * @return The friendly name of the adapter.
      */
     QString name() const;
 
     /**
-     * @return The alias of the adapter
+     * @return The system name (hostname) of the adapter.
      */
-    QString alias() const;
+    QString systemName() const;
 
     /**
      * @return The class of the adapter.
@@ -147,9 +152,9 @@ public:
 
 public Q_SLOTS:
     /**
-     *  Set the alias of the adapter
+     *  Set the name (alias) of the adapter
      */
-    void setAlias(const QString &alias);
+    void setName(const QString &name);
 
     /**
      * Sets whether this adapter is consuming energy or not.
-- 
2.1.0

